HACK.md

```diff
@@ -71,53 +70,65 @@ When building unit-tests, try to follow:
 make test
 ```
 
-## End-to-end tests
+## End-to-End Tests
 
 The following is a list of environment variables you can use when running e2e tests, this will override specific paths under the **Build** CRD [examples](samples/build).
 
-Env var | Path | Definition
---- | --- | --- |
-**TEST_IMAGE_REPO** | **spec.output.image** | Registry endpoint to push images |
-**TEST_IMAGE_REPO_SECRET** | **spec.output.credentials.name** | Registry endpoint secret(_usually of the type kubernetes.io/dockerconfigjson_) |
-
-For running E2E tests for private repositories, the **TEST_WITH_PRIVATE_REPO** environment variable needs to be set to **true**.
-If the Build private repositories [examples](test/data) contain references to private repositories you don´t have access, use
-the following variables for any related modification in the examples.
-
-Env var | Path | Definition
---- | --- | --- |
-**TEST_PRIVATE_REPO** | _none_ | Enables running e2e tests for private repositories |
-**TEST_PRIVATE_GITHUB** | **spec.source.url** | Private URL for the samples of the form *git@github.com* |
-**TEST_PRIVATE_GITLAB** | **spec.source.url** | Private URL for the samples of the form *git@gitlab.com* |
-**TEST_SOURCE_SECRET** | **spec.source.credentials.name** | A secret containing the SSH private key for accessing the above private repository. See [ssh-authentication](https://github.com/tektoncd/pipeline/blob/master/docs/auth.md#ssh-authentication-git). The secret definition must define two annotations: `tekton.dev/git-0: github.com` and `tekton.dev/git-1: gitlab.com`  |
+| Environment Variable     | Path                           | Description                           |
+|--------------------------|--------------------------------|---------------------------------------|
+| `TEST_IMAGE_REPO`        | `spec.output.image`            | Image repository for end-to-end tests |
+| `TEST_IMAGE_REPO_SECRET` | `spec.output.credentials.name` | Container credentials, JSON payload   |
+
+The contents of `TEST_IMAGE_REPO_SECRET` contents can be obtained from quay.io using a [robot account](https://docs.quay.io/glossary/robot-accounts.html). The contents of this file are like the following example:
+
+```json
+{
+  "auths": {
+    "quay.io": {
+      "auth": "<secret-credentials>",
+      "email": ""
+    }
+  }
+}
+```
```

Comment:
Just underlying that I am also against breaking the existing behavior of the `TEST_IMAGE_REPO_SECRET` environment variable which is supposed to contain the name of a credential. Extending the functionality to optionally support auto-generation is fine with me, the way described by @sbose78 [here](https://github.com/redhat-developer/build/pull/120#issuecomment-614883356) sounds good.

Reply:
Sure @SaschaSchwarze0, I'm going to change it back to original behaviour. Also, @qu1queee shares the same opinion. Duly noted.